src/quantity.rs

```diff
@@ -210,6 +210,35 @@ macro_rules! quantity {
             impl super::Conversion<V> for super::$unit {})+
         }
 
+        mod str {
+            storage_types! {
+                types: All;
+
+                use $crate::lib::str::FromStr;
+                use $crate::str::ParseQuantityError::*;
+
+                impl<U> FromStr for super::super::$quantity<U, V>
+                where
+                    U: super::super::super::Units<V> + ?Sized,
+                {
+                    type Err = $crate::str::ParseQuantityError;
+                    #[allow(unreachable_patterns)]
+                    fn from_str(s: &str) -> Result<Self, Self::Err> {
+                        let mut parts = s.splitn(2, ' ');
+                        let value = parts.next().unwrap();
```

Comment:
I don't like the `unwrap()`, but I haven't thought of a better way to handle this since I'm pretty sure that `splitn` will always return at least one group.

Reply:
Yep, I believe it does. Short of using `collect` and subscripts, I don't think there's a way to express this without `unwrap`.